File: java-app/src/main/java/com/mercadolibre/java_app/models/shipping/ShippingOption.java
package com.mercadolibre.java_app.models.shipping;

import android.os.Parcelable;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.mercadolibre.java_app.commons.ShippingOptionView;

public interface ShippingOption extends Parcelable {
    void showShippingPrices(@NonNull ShippingOptionView view);
    String calculatePrice(Double productPrices);
    ShippingOption getBestShippingOption(@Nullable ShippingOption otherShippingOption);
    boolean isBetterPriceThan(Double price);
}

File: kt-app/src/main/java/com/mercadolibre/melikotlinmeetup/commons/ShippingOption.kt
package com.mercadolibre.melikotlinmeetup.commons

import android.os.Parcelable

interface ShippingOption : Parcelable {
    fun showShippingPrices(view: ShippingOptionView)
    fun calculatePrice(productPrices: Double) : String
    fun getBestShippingOption(otherShippingOption: ShippingOption?): ShippingOption
    fun isBetterPriceThan(price: Double): Boolean
}